### ISSUE TITLE
Faster RCNN running evaluation failed when training

### ISSUE THREAD

**Issue by narutobns** [tensorpack/tensorpack#1039]
If you're asking about an unexpected problem you met, use this template.
__PLEASE DO NOT DELETE THIS TEMPLATE, FILL IT__:

### 1. What you did:

(1) **If you're using examples, what's the command you run:**
        ./train.py --config \
            MODE_MASK=True MODE_FPN=True \
            BACKBONE.WEIGHTS=../../../ImageNet-R101-AlignPadding.npz
(2) **If you're using examples, have you made any changes to the examples? Paste them here:**
       config.py
       83 _C.DATA.BASEDIR = '../../../COCO/DIR'
       85 _C.DATA.TRAIN = ['train2017']   # i.e. trainval35k, AKA train2017
       87 _C.DATA.VAL = ('val2017', )  # AKA val2017
       dataset.py
       18     class COCODetection(object):
       19     # handle the weird (but standard) split of train and val
       20     _INSTANCE_TO_BASEDIR = {
       21         'train2017': 'train2017',
       22         'val2017': 'val2017',
       23     }
(3) **If not using examples, tell us what you did here:**

Note that we may not be able to investigate it if there is no reproducible code.
It's always better to paste what you did instead of describing them.

### 2. What you observed:

(1) **Include the ENTIRE logs here:**

It's always better to paste what you observed instead of describing them.

A part of logs is sometimes enough, but it's always better to paste as much as possible.

You can run a command with `CMD 2>&1 | tee logs.txt` to save all stdout & stderr logs to one file.

[0109 16:21:54 @eval.py:255] Running evaluation ...
2019-01-09 16:21:55.400034: W tensorflow/core/kernels/queue_base.cc:277] _0_QueueInput/input_queue: Skipping cancelled enqueue attempt with queue not closed
Traceback (most recent call last):
  File "./train.py", line 514, in <module>
[0109 16:21:55 @input_source.py:176] EnqueueThread QueueInput/input_queue Exited.
    launch_train_with_config(traincfg, trainer)
  File "/home/user/anaconda3/envs/tf35/lib/python3.5/site-packages/tensorpack-0.9.0.1-py3.5.egg/tensorpack/train/interface.py", line 94, in launch_train_with_config
    extra_callbacks=config.extra_callbacks)
  File "/home/user/anaconda3/envs/tf35/lib/python3.5/site-packages/tensorpack-0.9.0.1-py3.5.egg/tensorpack/train/base.py", line 343, in train_with_defaults
    steps_per_epoch, starting_epoch, max_epoch)
  File "/home/user/anaconda3/envs/tf35/lib/python3.5/site-packages/tensorpack-0.9.0.1-py3.5.egg/tensorpack/train/base.py", line 315, in train
    self.main_loop(steps_per_epoch, starting_epoch, max_epoch)
  File "/home/user/anaconda3/envs/tf35/lib/python3.5/site-packages/tensorpack-0.9.0.1-py3.5.egg/tensorpack/utils/argtools.py", line 176, in wrapper
    return func(*args, **kwargs)
  File "/home/user/anaconda3/envs/tf35/lib/python3.5/site-packages/tensorpack-0.9.0.1-py3.5.egg/tensorpack/train/base.py", line 287, in main_loop
    self._callbacks.trigger_epoch()
  File "/home/user/anaconda3/envs/tf35/lib/python3.5/site-packages/tensorpack-0.9.0.1-py3.5.egg/tensorpack/callbacks/base.py", line 150, in trigger_epoch
    self._trigger_epoch()
  File "/home/user/anaconda3/envs/tf35/lib/python3.5/site-packages/tensorpack-0.9.0.1-py3.5.egg/tensorpack/callbacks/group.py", line 97, in _trigger_epoch
    cb.trigger_epoch()
  File "/home/user/anaconda3/envs/tf35/lib/python3.5/site-packages/tensorpack-0.9.0.1-py3.5.egg/tensorpack/callbacks/base.py", line 150, in trigger_epoch
    self._trigger_epoch()
  File "/home/user/hwy/tensorpack-master/examples/FasterRCNN/eval.py", line 256, in _trigger_epoch
    self._eval()
  File "/home/user/hwy/tensorpack-master/examples/FasterRCNN/eval.py", line 224, in _eval
    all_results = multithread_predict_dataflow(self.dataflows, self.predictors)
  File "/home/user/hwy/tensorpack-master/examples/FasterRCNN/eval.py", line 163, in multithread_predict_dataflow
    with ThreadPoolExecutor(max_workers=num_worker, thread_name_prefix='EvalWorker') as executor, \
TypeError: __init__() got an unexpected keyword argument 'thread_name_prefix'
MultiProcessMapDataZMQ successfully cleaned-up.
(2) **Other observations, if any:**
For example, CPU/GPU utilization, output images, tensorboard curves, if relevant to your issue.

### 3. What you expected, if not obvious.

If you expect higher accuracy, only in one of the two conditions can we help with it:
(1) You're unable to match the accuracy documented in tensorpack examples.
(2) It appears to be a tensorpack bug.

Otherwise, how to get high accuracy is a machine learning question and is
not our responsibility to figure out.

### 4. Your environment:
  + Python version:
    3.5.4
  + TF version: `python -c 'import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)'`.
    1.10.0
  + Tensorpack version: `python -c 'import tensorpack; print(tensorpack.__version__);'`.
      You can install Tensorpack master by `pip install -U git+https://github.com/ppwwyyxx/tensorpack.git`
      and see if your issue is already solved.
      0.9.0.1
  + If you're not using tensorpack under a normal command line shell (e.g.,
    using an IDE or jupyter notebook), please retry under a normal command line shell. 
  + Hardware information, e.g. number of GPUs used.
     2 GeForce GTX 1080 Ti
About efficiency issues, PLEASE first read http://tensorpack.readthedocs.io/en/latest/tutorial/performance-tuning.html

**Comment by ppwwyyxx** [comment]
Thanks for reporting! The code uses the `thread_name_prefix` argument which is available only after Python 3.6. Removing that argument will work

**Comment by ppwwyyxx** [comment]
( btw, this change is not needed:
```
20 _INSTANCE_TO_BASEDIR = {
21 'train2017': 'train2017',
22 'val2017': 'val2017',
23 }
```
because by default it will use the same name
